diff --git a/configure b/configure
index 9bcd60b..5efad0d 100755
--- a/configure
+++ b/configure
@@ -18716,9 +18716,9 @@ then
     as_fn_error $? "Please install zlib and zlib-devel packages" "$LINENO" 5
 else
 
-    vuln=`grep "ZLIB_VERSION \"1.2.0" $ZLIB_HOME/include/zlib.h`
+    vuln=`grep "ZLIB_VERSION \"1.2.0\"" $ZLIB_HOME/include/zlib.h`
     if test -z "$vuln"; then
-	vuln=`grep "ZLIB_VERSION \"1.2.1" $ZLIB_HOME/include/zlib.h`
+	vuln=`grep "ZLIB_VERSION \"1.2.1\"" $ZLIB_HOME/include/zlib.h`
     fi
 
     if test -n "$vuln"; then
