diff -r 8a6106504265 src/include/spf.h
--- a/src/include/spf.h	Thu Nov 03 01:35:30 2016 -0500
+++ b/src/include/spf.h	Thu Nov 03 03:52:05 2016 -0500
@@ -71,13 +71,13 @@
 /* ********************************************************************* */
 
 /** SPF_strerror() translates the SPF error number into a readable string */
-const char *SPF_strerror( SPF_errcode_t spf_err );
+const char * SPF_strerror( SPF_errcode_t spf_err );
 
 
 /** SPF_strresult() translates the SPF result number into a readable string */
-const char *SPF_strresult( SPF_result_t result );
+const char * SPF_strresult( SPF_result_t result );
 /** SPF_strreason() translates the SPF reason number into a readable string */
-const char *SPF_strreason( SPF_reason_t reason );
+const char * SPF_strreason( SPF_reason_t reason );
 
 /**
  * Returns the version information library. Useful if the library
@@ -86,6 +86,6 @@
  */
 void SPF_get_lib_version( int *major, int *minor, int *patch );
 
-const char *SPF_strrrtype(ns_type rr_type);
+const char * SPF_strrrtype(ns_type rr_type);
 
 #endif
diff -r 8a6106504265 src/include/spf_dns.h
--- a/src/include/spf_dns.h	Thu Nov 03 01:35:30 2016 -0500
+++ b/src/include/spf_dns.h	Thu Nov 03 03:52:05 2016 -0500
@@ -117,7 +117,7 @@
  */
 
 typedef void (*SPF_dns_destroy_t)(SPF_dns_server_t *spf_dns_server);
-typedef SPF_dns_rr_t *(*SPF_dns_lookup_t)(
+typedef SPF_dns_rr_t * (*SPF_dns_lookup_t)(
 				SPF_dns_server_t *spf_dns_server,
 				const char *domain,
 				ns_type ns_type, int should_cache
@@ -147,7 +147,7 @@
     SPF_dns_add_cache_t  add_cache;
 
     /* the next DNS layer down to call if this layer can't give an answer */
-    SPF_dns_server_t	*layer_below;
+    SPF_dns_server_t *layer_below;
 
     const char			*name;		/* name of the layer		*/
 	int					 debug;
@@ -156,13 +156,13 @@
 
 
 void			 SPF_dns_free( SPF_dns_server_t *spf_dns_server );
-SPF_dns_rr_t	*SPF_dns_lookup( SPF_dns_server_t *spf_dns_server,
+SPF_dns_rr_t	* SPF_dns_lookup( SPF_dns_server_t *spf_dns_server,
 			      const char *domain, ns_type rr_type,
 			      int should_cache );
-SPF_dns_rr_t	*SPF_dns_rlookup( SPF_dns_server_t *spf_dns_server,
+SPF_dns_rr_t	* SPF_dns_rlookup( SPF_dns_server_t *spf_dns_server,
 			       struct in_addr ipv4, ns_type rr_type,
 			       int should_cache );
-SPF_dns_rr_t	*SPF_dns_rlookup6( SPF_dns_server_t *spf_dns_server,
+SPF_dns_rr_t	* SPF_dns_rlookup6( SPF_dns_server_t *spf_dns_server,
 				struct in6_addr ipv6, ns_type rr_type,
 				int should_cache );
 
@@ -174,7 +174,7 @@
  * client IP address.  The complete validation procedure is described
  * in section 5.4 of the SPF spec.
  */
-char		*SPF_dns_get_client_dom(SPF_dns_server_t *spf_dns_server,
+char		* SPF_dns_get_client_dom(SPF_dns_server_t *spf_dns_server,
 				SPF_request_t *sr);
 
 
diff -r 8a6106504265 src/include/spf_dns_cache.h
--- a/src/include/spf_dns_cache.h	Thu Nov 03 01:35:30 2016 -0500
+++ b/src/include/spf_dns_cache.h	Thu Nov 03 03:52:05 2016 -0500
@@ -54,7 +54,7 @@
  * If debugging is turned on, information about cache hits and misses
  * will be generated.
  */
-SPF_dns_server_t	*SPF_dns_cache_new(SPF_dns_server_t *layer_below,
+SPF_dns_server_t	* SPF_dns_cache_new(SPF_dns_server_t *layer_below,
 				const char *name, int debug, int cache_bits);
 
 
diff -r 8a6106504265 src/include/spf_dns_null.h
--- a/src/include/spf_dns_null.h	Thu Nov 03 01:35:30 2016 -0500
+++ b/src/include/spf_dns_null.h	Thu Nov 03 03:52:05 2016 -0500
@@ -48,7 +48,7 @@
  * generated the output.
  */
  
-SPF_dns_server_t	*SPF_dns_null_new( SPF_dns_server_t *layer_below,
+SPF_dns_server_t	* SPF_dns_null_new( SPF_dns_server_t *layer_below,
 				const char *name, int debug);
 
 
diff -r 8a6106504265 src/include/spf_dns_resolv.h
--- a/src/include/spf_dns_resolv.h	Thu Nov 03 01:35:30 2016 -0500
+++ b/src/include/spf_dns_resolv.h	Thu Nov 03 03:52:05 2016 -0500
@@ -40,7 +40,7 @@
  * libresolv will be displayed.  This information is often not passed
  * on to (and not needed by) the higher level DNS layers.
  */
-SPF_dns_server_t	*SPF_dns_resolv_new(SPF_dns_server_t *layer_below,
+SPF_dns_server_t	* SPF_dns_resolv_new(SPF_dns_server_t *layer_below,
 				const char *name, int debug);
 
 #endif
diff -r 8a6106504265 src/include/spf_dns_rr.h
--- a/src/include/spf_dns_rr.h	Thu Nov 03 01:35:30 2016 -0500
+++ b/src/include/spf_dns_rr.h	Thu Nov 03 03:52:05 2016 -0500
@@ -70,13 +70,13 @@
     SPF_dns_server_t	*source;	/**< Which layer created this RR.  */
 } SPF_dns_rr_t;
 
-SPF_dns_rr_t	*SPF_dns_rr_new(void);
+SPF_dns_rr_t	* SPF_dns_rr_new(void);
 void			 SPF_dns_rr_free(SPF_dns_rr_t *spfrr);
-SPF_dns_rr_t	*SPF_dns_rr_new_init(SPF_dns_server_t *spf_dns_server,
+SPF_dns_rr_t	* SPF_dns_rr_new_init(SPF_dns_server_t *spf_dns_server,
 						const char *domain,
 						ns_type rr_type, int ttl,
 						SPF_dns_stat_t herrno);
-SPF_dns_rr_t	*SPF_dns_rr_new_nxdomain(SPF_dns_server_t *spf_dns_server,
+SPF_dns_rr_t	* SPF_dns_rr_new_nxdomain(SPF_dns_server_t *spf_dns_server,
 						const char *domain);
 
 SPF_errcode_t	 SPF_dns_rr_buf_realloc(SPF_dns_rr_t *spfrr,
diff -r 8a6106504265 src/include/spf_dns_test.h
--- a/src/include/spf_dns_test.h	Thu Nov 03 01:35:30 2016 -0500
+++ b/src/include/spf_dns_test.h	Thu Nov 03 03:52:05 2016 -0500
@@ -43,7 +43,7 @@
  * malloc'ed data, so they must be destroyed when you are finished
  * with them, or you will leak memory. 
  */
-SPF_dns_server_t	*SPF_dns_test_new( SPF_dns_server_t *layer_below,
+SPF_dns_server_t	* SPF_dns_test_new( SPF_dns_server_t *layer_below,
 				const char *name, int debug);
 
 #endif
diff -r 8a6106504265 src/include/spf_dns_zone.h
--- a/src/include/spf_dns_zone.h	Thu Nov 03 01:35:30 2016 -0500
+++ b/src/include/spf_dns_zone.h	Thu Nov 03 03:52:05 2016 -0500
@@ -67,7 +67,7 @@
  * that you can tell which zone layer in the stack of DNS layers
  * generated the output.
  */
-SPF_dns_server_t	*SPF_dns_zone_new(SPF_dns_server_t *layer_below,
+SPF_dns_server_t	* SPF_dns_zone_new(SPF_dns_server_t *layer_below,
 				const char *name, int debug);
 
 /**
diff -r 8a6106504265 src/include/spf_record.h
--- a/src/include/spf_record.h	Thu Nov 03 01:35:30 2016 -0500
+++ b/src/include/spf_record.h	Thu Nov 03 03:52:05 2016 -0500
@@ -225,7 +225,7 @@
 
 
 /** In spf_record.c */
-SPF_record_t	*SPF_record_new(SPF_server_t *spf_server,
+SPF_record_t * SPF_record_new(SPF_server_t *spf_server,
 			const char *text);
 void			 SPF_record_free(SPF_record_t *rp);
 void			 SPF_macro_free(SPF_macro_t *mac);
diff -r 8a6106504265 src/include/spf_request.h
--- a/src/include/spf_request.h	Thu Nov 03 01:35:30 2016 -0500
+++ b/src/include/spf_request.h	Thu Nov 03 03:52:05 2016 -0500
@@ -50,7 +50,7 @@
 	const char		*cur_dom;		/* "current domain" of SPF spec */
 };
 
-SPF_request_t	*SPF_request_new(SPF_server_t *spf_server);
+SPF_request_t	* SPF_request_new(SPF_server_t *spf_server);
 void			 SPF_request_free(SPF_request_t *sr);
 SPF_errcode_t	 SPF_request_set_ipv4(SPF_request_t *sr,
 						struct in_addr addr);
@@ -64,9 +64,9 @@
 						const char *dom);
 int				 SPF_request_set_env_from(SPF_request_t *sr,
 						const char *from);
-const char		*SPF_request_get_rec_dom(SPF_request_t *sr);
+const char		* SPF_request_get_rec_dom(SPF_request_t *sr);
 
-const char		*SPF_request_get_client_dom(SPF_request_t *sr);
+const char		* SPF_request_get_client_dom(SPF_request_t *sr);
 int				 SPF_request_is_loopback(SPF_request_t *sr);
 
 SPF_errcode_t	 SPF_request_query_mailfrom(SPF_request_t *spf_request,
diff -r 8a6106504265 src/include/spf_response.h
--- a/src/include/spf_response.h	Thu Nov 03 01:35:30 2016 -0500
+++ b/src/include/spf_response.h	Thu Nov 03 03:52:05 2016 -0500
@@ -169,7 +169,7 @@
 struct SPF_response_struct {
 	/* Structure variables */
 	SPF_request_t	*spf_request;
-	SPF_record_t	*spf_record_exp;
+	SPF_record_t *spf_record_exp;
 
 	/* The answer itself. */
 	SPF_result_t	 result;
@@ -193,20 +193,20 @@
 };
 
 
-SPF_response_t	*SPF_response_new(SPF_request_t *spf_request);
+SPF_response_t	* SPF_response_new(SPF_request_t *spf_request);
 void			 SPF_response_free(SPF_response_t *rp);
-SPF_response_t	*SPF_response_combine(SPF_response_t *main,
+SPF_response_t	* SPF_response_combine(SPF_response_t *main,
 					SPF_response_t *r2mx);
 
 	/* Query functions for elements of the result */
 SPF_result_t	 SPF_response_result(SPF_response_t *rp);
 SPF_reason_t	 SPF_response_reason(SPF_response_t *rp);
 SPF_errcode_t	 SPF_response_errcode(SPF_response_t *rp);
-const char		*SPF_response_get_received_spf(SPF_response_t *rp);
-const char		*SPF_response_get_received_spf_value(SPF_response_t *rp);
-const char		*SPF_response_get_header_comment(SPF_response_t *rp);
-const char		*SPF_response_get_smtp_comment(SPF_response_t *rp);
-const char		*SPF_response_get_explanation(SPF_response_t *rp);
+const char * SPF_response_get_received_spf(SPF_response_t *rp);
+const char * SPF_response_get_received_spf_value(SPF_response_t *rp);
+const char * SPF_response_get_header_comment(SPF_response_t *rp);
+const char * SPF_response_get_smtp_comment(SPF_response_t *rp);
+const char * SPF_response_get_explanation(SPF_response_t *rp);
 
 	/** How many warnings were generated? */
 int				 SPF_response_messages(SPF_response_t *rp);
@@ -215,7 +215,7 @@
 	/** Errors + warnings */
 int				 SPF_response_warnings(SPF_response_t *rp);
 	/** Returns an individual message */
-SPF_error_t		*SPF_response_message(SPF_response_t *rp, int idx);
+SPF_error_t		*         SPF_response_message(SPF_response_t *rp, int idx);
 
 SPF_errcode_t	 SPF_error_code(SPF_error_t *err);
 const char *	 SPF_error_message(SPF_error_t *err);
diff -r 8a6106504265 src/include/spf_server.h
--- a/src/include/spf_server.h	Thu Nov 03 01:35:30 2016 -0500
+++ b/src/include/spf_server.h	Thu Nov 03 03:52:05 2016 -0500
@@ -53,7 +53,7 @@
 #endif
 
 struct SPF_server_struct {
-	SPF_dns_server_t*resolver;		/**< SPF DNS resolver. */
+	SPF_dns_server_t *resolver;		/**< SPF DNS resolver. */
 	SPF_record_t	*local_policy;	/**< Local policies. */
 	SPF_macro_t		*explanation;	/**< Explanation string. */
 
@@ -73,8 +73,8 @@
 	SPF_DNS_RESOLV, SPF_DNS_CACHE, SPF_DNS_ZONE
 } SPF_server_dnstype_t;
 
-SPF_server_t	*SPF_server_new(SPF_server_dnstype_t dnstype,int debug);
-SPF_server_t	*SPF_server_new_dns(SPF_dns_server_t *dns,int debug);
+SPF_server_t	* SPF_server_new(SPF_server_dnstype_t dnstype,int debug);
+SPF_server_t	* SPF_server_new_dns(SPF_dns_server_t *dns,int debug);
 void			 SPF_server_free(SPF_server_t *sp);
 SPF_errcode_t	 SPF_server_set_rec_dom(SPF_server_t *sp,
 					const char *dom);
diff -r 8a6106504265 src/libspf2/spf_compile.c
--- a/src/libspf2/spf_compile.c	Thu Nov 03 01:35:30 2016 -0500
+++ b/src/libspf2/spf_compile.c	Thu Nov 03 03:52:05 2016 -0500
@@ -1198,7 +1198,7 @@
 	const char			*p;
 	int					 i;
 
-	const SPF_mechtype_t*redirect_mechtype = NULL;
+	const SPF_mechtype_t *redirect_mechtype = NULL;
 	int					 redirect_prefix;
 	const char			*redirect_val_start;
 
