diff -r 4630b883d8d3 src/include/spf_log.h
--- a/src/include/spf_log.h	Thu Nov 03 01:22:06 2016 -0500
+++ b/src/include/spf_log.h	Thu Nov 03 01:25:16 2016 -0500
@@ -60,17 +60,17 @@
 
 #if defined( __STDC_VERSION__ ) && __STDC_VERSION__ >= 199901L
 
-#define SPF_errorf(format, ... ) SPF_errorx( __FILE__, __LINE__, format, __VA_ARGS__ )
-#define SPF_warningf(format, ... ) SPF_warningx( __FILE__, __LINE__, format, __VA_ARGS__ )
-#define SPF_infof(format, ... ) SPF_infox( __FILE__, __LINE__, format, __VA_ARGS__ )
-#define SPF_debugf(format, ... ) SPF_debugx( __FILE__, __LINE__, format, __VA_ARGS__ )
+#define SPF_errorf(...) SPF_errorx( __FILE__, __LINE__, __VA_ARGS__ )
+#define SPF_warningf(...) SPF_warningx( __FILE__, __LINE__, __VA_ARGS__ )
+#define SPF_infof(...) SPF_infox( __FILE__, __LINE__, __VA_ARGS__ )
+#define SPF_debugf(...) SPF_debugx( __FILE__, __LINE__, __VA_ARGS__ )
 
 #elif defined( __GNUC__ )
 
-#define SPF_errorf(format... ) SPF_errorx( __FILE__, __LINE__, format )
-#define SPF_warningf(format... ) SPF_warningx( __FILE__, __LINE__, format )
-#define SPF_infof(format... ) SPF_infox( __FILE__, __LINE__, format )
-#define SPF_debugf(format... ) SPF_debugx( __FILE__, __LINE__, format )
+#define SPF_errorf(format, ...) SPF_errorx( __FILE__, __LINE__, format, ##__VA_ARGS__ )
+#define SPF_warningf(format, ...) SPF_warningx( __FILE__, __LINE__, format, ##__VA_ARGS__ )
+#define SPF_infof(format, ...) SPF_infox( __FILE__, __LINE__, format, ##__VA_ARGS__ )
+#define SPF_debugf(format, ...) SPF_debugx( __FILE__, __LINE__, format, ##__VA_ARGS__ )
 
 #else
 
