diff -r 6195942587b5 jansson/src/jansson.h
--- jansson/src/jansson.h	Fri Apr 29 18:42:11 2011 -0500
+++ jansson.b/src/jansson.h	Sat Apr 30 19:14:07 2011 -0500
@@ -86,24 +86,12 @@ json_t *json_true(void);
 json_t *json_false(void);
 json_t *json_null(void);

-static JSON_INLINE
-json_t *json_incref(json_t *json)
-{
-    if(json && json->refcount != (size_t)-1)
-        ++json->refcount;
-    return json;
-}
+json_t *json_incref(json_t *json);

 /* do not call json_delete directly */
 void json_delete(json_t *json);

-static JSON_INLINE
-void json_decref(json_t *json)
-{
-    if(json && json->refcount != (size_t)-1 && --json->refcount == 0)
-        json_delete(json);
-}
-
+void json_decref(json_t *json);

 /* error reporting */

@@ -135,23 +123,10 @@ const char *json_object_iter_key(void *i
 json_t *json_object_iter_value(void *iter);
 int json_object_iter_set_new(json_t *object, void *iter, json_t *value);

-static JSON_INLINE
-int json_object_set(json_t *object, const char *key, json_t *value)
-{
-    return json_object_set_new(object, key, json_incref(value));
-}

-static JSON_INLINE
-int json_object_set_nocheck(json_t *object, const char *key, json_t *value)
-{
-    return json_object_set_new_nocheck(object, key, json_incref(value));
-}
-
-static JSON_INLINE
-int json_object_iter_set(json_t *object, void *iter, json_t *value)
-{
-    return json_object_iter_set_new(object, iter, json_incref(value));
-}
+int json_object_set(json_t *object, const char *key, json_t *value);
+int json_object_set_nocheck(json_t *object, const char *key, json_t *value);
+int json_object_iter_set(json_t *object, void *iter, json_t *value);

 size_t json_array_size(const json_t *array);
 json_t *json_array_get(const json_t *array, size_t index);
@@ -162,23 +137,9 @@ int json_array_remove(json_t *array, siz
 int json_array_clear(json_t *array);
 int json_array_extend(json_t *array, json_t *other);

-static JSON_INLINE
-int json_array_set(json_t *array, size_t index, json_t *value)
-{
-    return json_array_set_new(array, index, json_incref(value));
-}
-
-static JSON_INLINE
-int json_array_append(json_t *array, json_t *value)
-{
-    return json_array_append_new(array, json_incref(value));
-}
-
-static JSON_INLINE
-int json_array_insert(json_t *array, size_t index, json_t *value)
-{
-    return json_array_insert_new(array, index, json_incref(value));
-}
+int json_array_set(json_t *array, size_t index, json_t *value);
+int json_array_append(json_t *array, json_t *value);
+int json_array_insert(json_t *array, size_t index, json_t *value);

 const char *json_string_value(const json_t *string);
 json_int_t json_integer_value(const json_t *integer);
diff -r 6195942587b5 jansson/src/value.c
--- jansson/src/value.c	Fri Apr 29 18:42:11 2011 -0500
+++ jansson.b/src/value.c	Sat Apr 30 19:14:07 2011 -0500
@@ -978,3 +978,49 @@ json_t *json_deep_copy(json_t *json)

     return NULL;
 }
+
+json_t *json_incref(json_t *json)
+{
+    if(json && json->refcount != (size_t)-1)
+        ++json->refcount;
+    return json;
+}
+
+void json_decref(json_t *json)
+{
+    if(json && json->refcount != (size_t)-1 && --json->refcount == 0)
+        json_delete(json);
+}
+
+int json_object_set(json_t *object, const char *key, json_t *value)
+{
+    return json_object_set_new(object, key, json_incref(value));
+}
+
+
+int json_object_set_nocheck(json_t *object, const char *key, json_t *value)
+{
+    return json_object_set_new_nocheck(object, key, json_incref(value));
+}
+
+
+int json_object_iter_set(json_t *object, void *iter, json_t *value)
+{
+    return json_object_iter_set_new(object, iter, json_incref(value));
+}
+
+int json_array_set(json_t *array, size_t index, json_t *value)
+{
+    return json_array_set_new(array, index, json_incref(value));
+}
+
+int json_array_append(json_t *array, json_t *value)
+{
+    return json_array_append_new(array, json_incref(value));
+}
+
+int json_array_insert(json_t *array, size_t index, json_t *value)
+{
+    return json_array_insert_new(array, index, json_incref(value));
+}
+
