diff -up zlib-1.2.3/contrib/minizip/miniunz.c.pom zlib-1.2.3/contrib/minizip/miniunz.c
--- zlib-1.2.3/contrib/minizip/miniunz.c.pom	2005-07-12 20:08:40.000000000 +0200
+++ zlib-1.2.3/contrib/minizip/miniunz.c	2008-12-01 11:12:23.000000000 +0100
@@ -112,6 +112,11 @@ int makedir (newdir)
     return 0;
 
   buffer = (char*)malloc(len+1);
+  if (buffer==NULL)
+  {
+    printf("Error allocating memory\n");
+    return UNZ_INTERNALERROR;
+  }                          
   strcpy(buffer,newdir);
 
   if (buffer[len-1] == '/') {
