diff --git a/libtest/unittest.cc b/libtest/unittest.cc
index a2843d2..a1922db 100644
--- a/libtest/unittest.cc
+++ b/libtest/unittest.cc
@@ -618,14 +618,14 @@ static test_return_t application_doesnotexist_BINARY(void *)
   true_app.will_fail();
 
   const char *args[]= { "--fubar", 0 };
-#if defined(__APPLE__) && __APPLE__
-  ASSERT_EQ(Application::INVALID_POSIX_SPAWN, true_app.run(args));
-#elif defined(__FreeBSD__) && __FreeBSD__
-  ASSERT_EQ(Application::INVALID_POSIX_SPAWN, true_app.run(args));
-#else
-  ASSERT_EQ(Application::SUCCESS, true_app.run(args));
-  ASSERT_EQ(Application::INVALID_POSIX_SPAWN, true_app.join());
-#endif
+
+  Application::error_t rc =  true_app.run(args);
+  if (rc == Application::SUCCESS) {
+  	ASSERT_EQ(Application::INVALID_POSIX_SPAWN, true_app.join());
+  }
+  else {
+  	ASSERT_EQ(Application::INVALID_POSIX_SPAWN, rc);
+  }
 
   test_zero(true_app.stdout_result().size());
 
@@ -877,7 +877,7 @@ static test_return_t get_free_port_TEST(void *)
 static test_return_t fatal_TEST(void *)
 {
   ASSERT_EQ(fatal_calls++, fatal::disabled_counter());
-  throw libtest::fatal(LIBYATL_DEFAULT_PARAM, "Testing va_args based fatal(): %d", 10); 
+  throw libtest::fatal(LIBYATL_DEFAULT_PARAM, "Testing va_args based fatal(): %d", 10);
 
   return TEST_SUCCESS;
 }
