diff --git a/libtest/memcached.cc b/libtest/memcached.cc
index 0127356..a6817fe 100644
--- a/libtest/memcached.cc
+++ b/libtest/memcached.cc
@@ -82,6 +82,7 @@ public:
                     memcached_binary(), false, is_socket_arg)
   {
     set_pid_file();
+    set_log_file();
   }
 
   virtual const char *sasl() const
diff --git a/libtest/server.cc b/libtest/server.cc
index d024d05..2d5cc0c 100644
--- a/libtest/server.cc
+++ b/libtest/server.cc
@@ -443,7 +443,7 @@ bool Server::set_log_file()
   file_buffer.resize(FILENAME_MAX);
   file_buffer[0]= 0;
 
-  snprintf(&file_buffer[0], file_buffer.size(), "var/log/%s.logXXXXXX", name());
+  snprintf(&file_buffer[0], file_buffer.size(), "/tmp/%s.log_XXXXXX", name());
   int fd;
   if ((fd= mkstemp(&file_buffer[0])) == -1)
   {
diff --git a/libtest/server.h b/libtest/server.h
index bbebe95..ea9cf50 100644
--- a/libtest/server.h
+++ b/libtest/server.h
@@ -273,6 +273,7 @@ public:
 
 protected:
   bool set_pid_file();
+  bool set_log_file();
   Options _options;
   Application _app;
 
@@ -280,7 +281,6 @@ private:
   bool is_helgrind() const;
   bool is_valgrind() const;
   bool is_debug() const;
-  bool set_log_file();
   bool set_socket_file();
   void reset_pid();
   bool out_of_ban_killed_;
